File: lib/src/main/java/me/ameriod/lib/mvp/view/deligate/ControllerDelegate.java
package me.ameriod.lib.mvp.view.deligate;

import android.os.Bundle;
import android.support.annotation.NonNull;

import me.ameriod.lib.mvp.Mvp;

public interface ControllerDelegate<V extends Mvp.View, P extends Mvp.Presenter<V>> {

    void onCreateView();

    void onRestoreViewState(@NonNull Bundle savedViewState);

    void onSaveViewState(@NonNull Bundle outState);

    void onDestroyView();

    P getPresenter();
}

File: lib/src/main/java/me/ameriod/lib/mvp/deligate/ControllerDelegate.kt
package me.ameriod.lib.mvp.deligate

import android.os.Bundle

import me.ameriod.lib.mvp.Mvp

interface ControllerDelegate<in V : Mvp.View, out P : Mvp.Presenter<in V>> {

    fun onCreateView()

    fun onRestoreViewState(savedViewState: Bundle)

    fun onSaveViewState(outState: Bundle)

    fun onDestroyView()

    fun getPresenter(): P
}
